lib/pengine/native.c

```diff
@@ -545,16 +549,13 @@ native_print(resource_t * rsc, const char *pre_text, long options, void *print_d
 
         /* Ignore target role Started, as it is the default anyways
          * (and would also allow a Master to be Master).
-         * Show if current role differs from target role,
-         * or if target role limits our abilities. */
+         * Show if target role limits our abilities. */
         if (target_role_e == RSC_ROLE_STOPPED) {
             flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset, "%sdisabled", flagOffset?", ":"");
             rsc->cluster->disabled_resources++;
 
         } else if (uber_parent(rsc)->variant == pe_master
-                   && target_role_e > RSC_ROLE_STOPPED
-                   && target_role_e < RSC_ROLE_MASTER
-                   && safe_str_neq(target_role, role2text(rsc->role))) {
+                   && target_role_e == RSC_ROLE_SLAVE) {
```

Comment:
This one took me longer to get, but I think it's correct, too. The point of this ''if'' block is, if there's a target-role configured for a master-slave resource, do we need to print the target-role? The original code would only print if >stopped and <master (hence Started or Slave), so presumably we can consider just those. Again, Started doesn't constrain us in any way, so there's no need to print it (even if the resource is stopped, the target-role isn't preventing it from starting). By contrast, a target-role of Slave is always interesting, even if the instance if already a slave (equivalent to printing why a stopped resource is stopped).